pkg/logql/log/distinct_test.go

```diff
@@ -0,0 +1,61 @@
+package log
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/logqlmodel"
+)
+
+func Test_DistinctFilter(t *testing.T) {
+
+	c := struct {
+		name          string
+		label         []string
+		lbs           labels.Labels
+		input         []string
+		expectedCount int
+		expectedLines []string
+	}{
+		name:  "distinct test",
+		label: []string{"id", "time"},
```

Comment:
in case it makes sense: could you add to this list a label that won't be present on the given input?

Reply:
good point. After adding a label that does not exist, in order to make the test pass, I have to modify `false` to `true`, please help review whether this modification is correct

![image](https://user-images.githubusercontent.com/9583245/233527135-2565fd02-b13e-4379-be8b-f10b7171a93d.png)
![image](https://user-images.githubusercontent.com/9583245/233527161-17abfa2a-ca55-4e36-957d-11dfe35f4b5f.png)